### PR TITLE
Fix issue with Node installation

### LINKED ISSUE #3743: Error when running grpc on node
When running a module that requires grpc, I see the stack trace below. I installed grpc by running `npm install git://github.com/grpc/grpc.git`. The installation completes normally with no errors.

\<project_dir\>/node_modules/grpc/node_modules/bindings/bindings.js:91
  throw err
  ^

Error: Could not locate the bindings file. Tried:
 → \<project_dir\>/node_modules/grpc/build/grpc_node.node
 → \<project_dir\>/node_modules/grpc/build/Debug/grpc_node.node
 → \<project_dir\>/node_modules/grpc/build/Release/grpc_node.node
 → \<project_dir\>/node_modules/grpc/out/Debug/grpc_node.node
 → \<project_dir\>/node_modules/grpc/Debug/grpc_node.node
 → \<project_dir\>/node_modules/grpc/out/Release/grpc_node.node
 → \<project_dir\>/node_modules/grpc/Release/grpc_node.node
 → \<project_dir\>/node_modules/grpc/build/default/grpc_node.node
 → \<project_dir\>/node_modules/grpc/compiled/4.1.1/linux/x64/grpc_node.node
    at bindings (\<project_dir\>node_modules/grpc/node_modules/bindings/bindings.js:88:9)
    at Object.\<anonymous\> (\<project_dir\>/node_modules/grpc/src/node/src/client.js:43:31)
    at Module._compile (module.js:434:26)
    at Object.Module._extensions..js (module.js:452:10)
    at Module.load (module.js:355:32)
    at Function.Module._load (module.js:310:12)
    at Module.require (module.js:365:17)
    at require (module.js:384:17)
    at Object.\<anonymous\> (\<project_dir\>/node_modules/grpc/src/node/index.js:40:14)
    at Module._compile (module.js:434:26)

### PR DESCRIPTION
**Pull request by murgatroid99**
This fixes #3743 by consolidating the gyp files into a single file.

Also fixes some tests that broke somewhere along the way.

### CODE REVIEW

**Comment by ctiller** [comment]
Look at that beautiful tick!

File: templates/binding.gyp.template [hunk 1]
```diff
@@ -39,54 +39,24 @@
...
-          },
-        }
-      },
-      'msvs_settings': {
-        'VCLinkerTool': {
-          'GenerateDebugInformation': 'true',
-        },
-      },
-      # TODO: Add fallback for Windows, and if pkg-config is not available
+        # Emperically, Node only exports ALPN symbols if its major version is >0.
```
Comment (jtattermusch) [inline templates/binding.gyp.template]: Empirically

Comment (murgatroid99) [inline templates/binding.gyp.template]: Fixed

**Comment by jtattermusch** [comment]
Jenkins was green before fixing the typo, so merging.

**Pull request merged**